src/sio.ts

```diff
@@ -46,6 +46,18 @@ export class RPSIO {
 
   constructor(private readonly rp2040: RP2040) {}
 
+  updateHardwareDivider() {
+    if (this.divDivisor == 0) {
+      this.divQuotient = 0xffffffff;
+      this.divRemainder = this.divDividend;
```

Comment:
Shouldn't we update divCSR / cycles in this case as well?

Reply:
correct, sorry stupid bug.